src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx

```diff
@@ -428,6 +445,104 @@ export default function V2ProjectReconfigureModal({
     setRulesDrawerVisible(false)
   }
 
+  const fundingDrawerHasSavedChanges = () => {
+    const fundingCycleData = serializeV2FundingCycleData(
+      editingFundingCycleData,
+    )
+    const fundAccessConstraints = editingFundAccessConstraints.length
+      ? serializeFundAccessConstraint(editingFundAccessConstraints?.[0])
+      : undefined
+    const payoutGroupedSplits = editingPayoutGroupedSplits.splits
+
+    if (!fundAccessConstraints || !initialEditingData) {
+      return false
+    }
+    const durationUpdated =
+      fundingCycleData.duration !== initialEditingData.fundingCycleData.duration
+    const distributionLimitUpdated =
+      fundAccessConstraints.distributionLimit !==
+      initialEditingData.fundAccessConstraints?.[0].distributionLimit
+    const distributionLimitCurrencyUpdated =
+      fundAccessConstraints.distributionLimitCurrency !==
+      initialEditingData.fundAccessConstraints?.[0].distributionLimitCurrency
+    const payoutGroupedSplitsUpdated = !isEqual(
+      payoutGroupedSplits,
+      initialEditingData.payoutGroupedSplits?.payoutGroupedSplits ?? [],
+    )
+    return (
+      durationUpdated ||
+      distributionLimitUpdated ||
+      distributionLimitCurrencyUpdated ||
+      payoutGroupedSplitsUpdated
+    )
+  }
+
+  const tokenDrawerHasSavedChanges = () => {
+    const fundingCycleData = serializeV2FundingCycleData(
+      editingFundingCycleData,
+    )
+    const fundingCycleMetadata = serializeV2FundingCycleMetadata(
+      editingFundingCycleMetadata,
+    )
+    const fundAccessConstraints = editingFundAccessConstraints.length
+      ? serializeFundAccessConstraint(editingFundAccessConstraints?.[0])
+      : undefined
+    const reservedTokensGroupedSplits =
+      editingReservedTokensGroupedSplits.splits
+
+    if (!fundAccessConstraints || !initialEditingData) {
+      return false
+    }
+
+    const reservedRateUpdated =
+      fundingCycleMetadata.reservedRate !==
+      initialEditingData.fundingCycleMetadata.reservedRate
+    const reservedTokensGroupedSplitsUpdated = !isEqual(
+      reservedTokensGroupedSplits,
+      initialEditingData.payoutGroupedSplits.reservedTokensGroupedSplits ?? [],
+    )
+    const discountRateUpdated =
+      fundingCycleData.discountRate !==
+      initialEditingData.fundingCycleData.discountRate
+    const redemptionRateUpdated =
+      fundingCycleMetadata.redemptionRate !==
+      initialEditingData.fundingCycleMetadata.redemptionRate
+
+    return (
+      reservedRateUpdated ||
+      reservedTokensGroupedSplitsUpdated ||
+      discountRateUpdated ||
+      redemptionRateUpdated
+    )
+  }
+
+  const rulesDrawerHasSavedChanges = () => {
```

Comment:
**suggestion (non-blocking)**: at this point it might make sense to have 3 components for each of these buttons, and include this `hasSavedChanges` logic included in them. 

No big for now 🙏